lib/src/signer.rs

```diff
@@ -0,0 +1,51 @@
+#[cfg(feature = "signing-gpgme")]
+use gpgme::{Context, Key, Protocol};
+
+use crate::backend::BackendResult;
+
+pub trait Signer {
+    fn sign(&self, input: &[u8]) -> BackendResult<String>;
+}
+
+#[cfg(feature = "signing-gpgme")]
+pub struct GpgMeSigner {
+    key: Option<Key>,
+}
+
+#[cfg(feature = "signing-gpgme")]
+impl GpgMeSigner {
+    pub fn new(key_id: Option<String>) -> Self {
+        let key = match key_id {
+            Some(key_id) => {
+                let mut ctx =
+                    Context::from_protocol(Protocol::OpenPgp).expect("from_protocol failed");
+                let mut keys = ctx.find_keys([key_id]).expect("failed to find keys");
+                let key = keys
+                    .next()
+                    .expect("no key found")
+                    .expect("failed to get key");
+                Some(key)
+            }
+            None => None,
+        };
+        Self { key }
+    }
+}
+
+#[cfg(feature = "signing-gpgme")]
+impl Signer for GpgMeSigner {
+    fn sign(&self, input: &[u8]) -> BackendResult<String> {
+        let mut ctx = Context::from_protocol(Protocol::OpenPgp).unwrap();
```

Comment:
nit: We could avoid the unwrap (and save time?) by keeping the context object that we created earlier (by adding it in `GpgMeSigner`).